packages/scss/src/components/inputs/input/_input.framed.scss

```diff
@@ -0,0 +1,142 @@
+%input_framed {
+	padding: 0;
+	width: 100%;
+
+	.input-field {
+		@include fakeBorder(1px, _theme("commons.divider.color"));
+
+		height: 100%;
+		margin-bottom: 0;
+		padding: _component("input.framed.top-padding") _component("input.framed.side-padding") _component("input.framed.bottom-padding");
+		width: 100%;
+		&:focus, &:hover {
+			z-index: 1;
+			~ .input-messages {
+				transform: translateY(100%);
+			}
+		}
+	}
+
+	.input-label {
+		font-size: _theme("sizes.small.font-size");
+		left: _component("input.framed.side-padding");
+		right: _component("input.framed.side-padding");
+		top: _component("input.framed.label-top-offset");
+		width: auto;
+	}
+
+	.input-suffix {
+		bottom: _component("input.framed.bottom-padding");
+		color: _color("text.placeholder");
+		position: absolute;
+		right: _component("input.framed.side-padding");
+		text-align: right;
+	}
+
+	.input-messages {
+		bottom: 0;
+		left: 0;
+		pointer-events: none;
+		position: absolute;
+		right: 0;
+		text-align: center;
+		transform: translateY(0);
+		transition: transform _theme("commons.animations.durations.standard");
+		z-index: 1;
+
+		.input-error {
+			background-color: _color("error");
+			color: _theme("palettes.error.text");
+			font-size: _component("input.message.font-size");
+			margin: 0;
+		}
+
+		.input-warning {
+			background-color: _color("warning");
+			color: _theme("palettes.warning.text");
+			font-size: _component("input.message.font-size");
+			margin: 0;
+		}
+
+		.input-helper {
+			background-color: _color("text.light");
+			color: white;
+			font-size: _component("input.message.font-size");
+			margin: 0;
+		}
+	}
+
+	// COLORING
+	@mixin inputColoring($palette) {
+		.input-field:focus, .input-field:hover {
+			@include fakeBorder(1px, _get($palette, "color"));
+
+			background-color: transparentize(_get($palette, "color"), .95);
+			~ .input-label {
+				color: _get($palette, "color");
+			}
+		}
+	}
+
+	@include inputColoring(_component("input.framed.default-palette"));
+
+	// palette colors
+	@each $name, $palette in _palettes() {
+		&.palette-#{$name} {
+			@include inputColoring($palette);
+		}
+	}
+	// MOD
+	&.has-suffix {
+		.input-field {
+			padding-right: 3rem;
+		}
+	}
+
+	&.mod-select, &.mod-search {
+		&::after {
+			right: _component("input.framed.select.icon-offset");
+		}
+		.input-options {
+			top: _component("input.framed.select.options-offset");
+		}
+	}
+	// STATUS
+
+	.input-field {
+		&[disabled], &[readonly] {
+			background-color: transparentize(_color("text.placeholder"), .8);
+			background-image: none;
+			opacity: .6;
+			& + label {
```

Comment:
Should be a class name

Reply:
I'll add is-disabled and is-readonly to every input